flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala

```diff
@@ -33,6 +33,14 @@ object ScalarSqlFunctions {
     OperandTypes.NILADIC,
     SqlFunctionCategory.NUMERIC)
 
+  val BIN = new SqlFunction(
+    "BIN",
+    SqlKind.OTHER_FUNCTION,
+    ReturnTypes.explicit(SqlTypeName.VARCHAR),
+    null,
+    OperandTypes.NUMERIC,
```

Comment:
I don't think BIN accepts all NUMERIC operands. I think it only accepts BIGINT, TINYINT, SMALLINT,INTEGER.  And not support other decimal numeric.

Reply:
Yes, In current code we only support BYTE,SHORT,INTEGER,LONG. If we try to using a FLOAT input we will got the EXCEPTION. 
If understand you correctly you want define a new operand types checker, just like `tableAPI(ScalarSqlFunction)`, But I in SQL I'd better using CALCITE's default `OperandTypeChecker`. What do you think?

Reply:
You can construct a checker easily using `OperandTypes.or()` in Calcite.